tests/Exercise/TopScorersTest.elm

```diff
@@ -0,0 +1,206 @@
+                        [ TestHelper.createExpectedErrorUnder (Comment "Uses List.sort" "elm.top-scorers.dont_use_sort" Essential Dict.empty) "List.sort"
+                            |> Review.Test.atExactly { start = { row = 6, column = 26 }, end = { row = 6, column = 35 } }
+                        ]
+        , test "should report that Dict.merge must be used" <|
+            \() ->
+                """
+module TopScorers exposing (..)
+
+combineGames : Dict PlayerName Int -> Dict PlayerName Int -> Dict PlayerName Int
+combineGames game1 game2 =
+    Debug.todo "implement this function"
+            """
+                    |> Review.Test.run TopScorers.combineGamesMustUseMerge
+                    |> Review.Test.expectErrors
+                        [ TestHelper.createExpectedErrorUnder (Comment "Doesn't use Dict.merge" "elm.top-scorers.use_merge" Essential Dict.empty) "combineGames"
+                            |> Review.Test.atExactly { start = { row = 5, column = 1 }, end = { row = 5, column = 13 } }
+                        ]
+        , test "should report that List.foldl must be used" <|
+            \() ->
+                """
+module TopScorers exposing (..)
+
+aggregateScorers : List PlayerName -> Dict PlayerName Int
+aggregateScorers playerNames =
+    Debug.todo "implement this function"
+            """
+                    |> Review.Test.run TopScorers.aggregateScorersMustUseFoldl
+                    |> Review.Test.expectErrors
+                        [ TestHelper.createExpectedErrorUnder (Comment "Doesn't use List.foldl" "elm.top-scorers.use_foldl" Essential Dict.empty) "aggregateScorers"
+                            |> Review.Test.atExactly { start = { row = 5, column = 1 }, end = { row = 5, column = 17 } }
+                        ]
+        , test "should report that updateGoalCountForPlayer must be used" <|
+            \() ->
+                """
+module TopScorers exposing (..)
+
+aggregateScorers : List PlayerName -> Dict PlayerName Int
+aggregateScorers playerNames =
+    Debug.todo "implement this function"
+            """
+                    |> Review.Test.run TopScorers.aggregateScorersMustUseUpdateGoalCountForPlayer
+                    |> Review.Test.expectErrors
+                        [ TestHelper.createExpectedErrorUnder (Comment "Doesn't use updateGoalCountForPlayer" "elm.top-scorers.use_updateGoalCountForPlayer" Essential Dict.empty) "aggregateScorers"
+                            |> Review.Test.atExactly { start = { row = 5, column = 1 }, end = { row = 5, column = 17 } }
+                        ]
+        , test "should report that Maybe.map must be used" <|
+            \() ->
+                """
+module TopScorers exposing (..)
+
+formatPlayer : PlayerName -> Dict PlayerName Int -> String
+formatPlayer playerName playerGoalCounts =
+    Debug.todo "implement this function"
+            """
+                    |> Review.Test.run TopScorers.formatPlayerMustUseMap
```

Comment:
I can't remember if I tried to implement that, but do you think it would be useful to add a helper that tests more than one rule at a time? Or is it clearer to test them separately?

Reply:
After addressing your comments in the sister PR, I don't think this is relevant any more as both checks are now done in the same rule ...